spec/test.spec.js

```diff
@@ -323,76 +323,79 @@ describe('S3Adapter tests', () => {
-    });
-
-    it('should get without directAccess', () => {
-      options.directAccess = false;
-      const s3 = new S3Adapter('accessKey', 'secretKey', 'myBucket', options);
-      expect(s3.getFileLocation(testConfig, 'test.png')).toEqual('http://my.server.com/parse/files/xxxx/test.png');
-    });
-
-    it('should go directly to amazon', () => {
-      delete options.baseUrl;
-      const s3 = new S3Adapter('accessKey', 'secretKey', 'myBucket', options);
-      expect(s3.getFileLocation(testConfig, 'test.png')).toEqual('https://myBucket.s3.amazonaws.com/foo/bar/test.png');
-    });
-  });
-  describe('getFileLocation', () => {
-    const testConfig = {
-      mount: 'http://my.server.com/parse',
-      applicationId: 'xxxx',
-    };
-    let options;
-
-    beforeEach(() => {
-      options = {
-        directAccess: true,
-        bucketPrefix: 'foo/bar/',
-        baseUrl: (fileconfig, filename) => {
-          if (filename.length > 12) {
-            return 'http://example.com/files';
-          }
-          return 'http://example.com/files';
+    it('when use presigned URL should use S3 \'getObject\' operation', () => {
+      options.presignedUrl = true;
+      const s3 = new S3Adapter('accessKey', 'secretKey', 'my-bucket', options);
+      const originalS3Client = s3._s3Client;
+      let getSignedUrlOperation = '';
+      s3._s3Client = {
+        getSignedUrl: (operation, params, callback) => {
+          getSignedUrlOperation = operation;
+          return originalS3Client.getSignedUrl(operation, params, callback);
         },
       };
+
+      s3.getFileLocation(testConfig, 'test.png');
+      expect(getSignedUrlOperation).toBe('getObject');
     });
 
-    it('should get using the baseUrl', () => {
-      const s3 = new S3Adapter('accessKey', 'secretKey', 'myBucket', options);
-      expect(s3.getFileLocation(testConfig, 'test.png')).toEqual('http://example.com/files/foo/bar/test.png');
+    it('should get using the baseUrl and amazon using presigned URL', () => {
+      options.presignedUrl = true;
+      const s3 = new S3Adapter('accessKey', 'secretKey', 'my-bucket', options);
+
+      const fileLocation = s3.getFileLocation(testConfig, 'test.png');
+      expect(fileLocation).toMatch(/^http:\/\/example.com\/files\/foo\/bar\/test.png\?/);
+      expect(fileLocation).toMatch(/X-Amz-Credential=accessKey%2F\d{8}%2F\w{2}-\w{1,9}-\d%2Fs3%2Faws4_request/);
+      expect(fileLocation).toMatch(/X-Amz-Date=\d{8}T\d{6}Z/);
+      expect(fileLocation).toMatch(/X-Amz-Signature=.{64}/);
+      expect(fileLocation).toMatch(/X-Amz-Expires=\d{1,6}/);
+      expect(fileLocation).toContain('X-Amz-Algorithm=AWS4-HMAC-SHA256');
+      expect(fileLocation).toContain('X-Amz-SignedHeaders=host');
     });
 
     it('should get direct to baseUrl', () => {
       options.baseUrlDirect = true;
-      const s3 = new S3Adapter('accessKey', 'secretKey', 'myBucket', options);
+      const s3 = new S3Adapter('accessKey', 'secretKey', 'my-bucket', options);
```

Comment:
What's the reason for this change?

Reply:
going by the [ original authors commit](https://github.com/parse-community/parse-server-s3-adapter/commit/140bdf95c5372c6accdc6ad01e1f266f8b9fcd7b
) i think the change was to reflect s3 bucket naming rules
> Bucket names can consist only of lowercase letters, numbers, dots (.), and hyphens (-).
- https://docs.aws.amazon.com/AmazonS3/latest/userguide/bucketnamingrules.html

Reply:
Makes sense, let's keep this change then